### PR TITLE
Add workaround for table mutation issue

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
Add workaround for table mutation [issue](https://github.com/ballerina-platform/ballerina-lang/issues/36056)

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: compiler/modules/nback/module.bal [hunk 1]
```diff
@@ -102,10 +102,17 @@ function orderByFwdTargets(bir:BasicBlock[] blocks, bir:Label label, boolean[] v
...
+    return ty.index;
+}
+
 function createTypeUsage(table<UsedSemType> usedSemTypes) returns TypeUsage {
     byte[] uses = [];
     t:SemType[] types = [];
-    foreach var used in usedSemTypes {
+    // JBUG 36056 can't directly iterate over usedSemTypes
+    UsedSemType[] usedSemTy = usedSemTypes.toArray();
+    usedSemTy = usedSemTy.sort("ascending", index);
```
Comment (jclark) [inline compiler/modules/nback/module.bal]: I don't think you need this.  They are added to the table in order of increasing index. So the workaround can just change the next line to `usedSemTypes.toArray()`.

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/modules/nback/module.bal (hunk 1 above)
Comment (heshanpadmasiri) [reply compiler/modules/nback/module.bal]: I also thought the same but just for the failing testcase what I am getting from `usedSemTypes.toArray()` is not sorted.

Comment (jclark) [reply compiler/modules/nback/module.bal]: That should be reported as another bug (maybe the spec is not as clear as it should be here).

Comment (jclark) [inline compiler/modules/nback/module.bal]: How about `from var u in usedSemType select u`? That should give them in the right order.

Comment (heshanpadmasiri) [reply compiler/modules/nback/module.bal]: fixed (used `toArray()` function since directly using the query expression without it causes the same JBUG)

**Pull request merged**